server/admin/admin_server_test.go

```diff
@@ -34,39 +42,228 @@ func TestAdminServer_Unimplemented(t *testing.T) {
+		},
+		{
+			desc:   "unknownTree",
+			getErr: true,
+		},
+		{
+			desc:      "commitError",
+			commitErr: true,
+		},
+	}
+
+	ctx := context.Background()
+	for _, test := range tests {
+		setup := setupAdminServer(ctrl, true /* snapshot */, !test.getErr /* shouldCommit */, test.commitErr)
+		tx := setup.snapshotTX
+		s := setup.server
+
+		req := &trillian.GetTreeRequest{TreeId: 12345}
+		wantTree := testonly.LogTree
+		if test.getErr {
+			tx.EXPECT().GetTree(ctx, req.TreeId).Return(nil, errors.New("GetTree failed"))
+		} else {
+			tx.EXPECT().GetTree(ctx, req.TreeId).Return(wantTree, nil)
+		}
+		wantErr := test.getErr || test.commitErr
+
+		tree, err := s.GetTree(ctx, req)
+		if hasErr := err != nil; hasErr != wantErr {
+			t.Errorf("%v: GetTree() = (_, %v), wantErr = %v", test.desc, err, wantErr)
+			continue
+		} else if hasErr {
+			continue
+		}
+		if !reflect.DeepEqual(tree, wantTree) {
+			t.Errorf("%v: GetTree() = (%v, _), want = (%v, _)", test.desc, tree, wantTree)
+		}
+	}
+}
+
+func TestAdminServer_CreateTree(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	invalidTree := *testonly.LogTree
+	invalidTree.TreeState = trillian.TreeState_HARD_DELETED
+
+	tests := []struct {
+		desc                 string
+		req                  *trillian.CreateTreeRequest
+		createErr, commitErr bool
+	}{
+		{
+			desc: "validTree",
+			req:  &trillian.CreateTreeRequest{Tree: testonly.LogTree},
+		},
+		{
+			desc:      "invalidTree",
+			req:       &trillian.CreateTreeRequest{Tree: &invalidTree},
+			createErr: true,
+		},
+		{
+			desc:      "commitError",
+			req:       &trillian.CreateTreeRequest{Tree: testonly.LogTree},
+			commitErr: true,
+		},
+	}
+
+	ctx := context.Background()
+	for _, test := range tests {
+		setup := setupAdminServer(ctrl, false /* snapshot */, !test.createErr /* shouldCommit */, test.commitErr)
+		tx := setup.tx
+		s := setup.server
+
+		if test.createErr {
+			tx.EXPECT().CreateTree(ctx, test.req.Tree).Return(nil, errors.New("CreateTree failed"))
+		} else {
+			newTree := *test.req.Tree
+			tx.EXPECT().CreateTree(ctx, test.req.Tree).Return(&newTree, nil)
+		}
+		wantErr := test.createErr || test.commitErr
+
+		tree, err := s.CreateTree(ctx, test.req)
+		if hasErr := err != nil; hasErr != wantErr {
+			t.Errorf("%v: CreateTree() = (_, %v), wantErr = %v", test.desc, err, wantErr)
+			continue
+		} else if hasErr {
+			continue
+		}
+		if !reflect.DeepEqual(tree, test.req.Tree) {
+			t.Errorf("%v: CreateTree() = (%v, _), want = (%v, _)", test.desc, tree, test.req.Tree)
+		}
+	}
+}
+
+// adminTestSetup contains an operational adminServer and required dependencies.
+// It's created via setupAdminServer.
+type adminTestSetup struct {
+	registry   *extension.MockRegistry
+	as         *storage.MockAdminStorage
+	tx         *storage.MockAdminTX
+	snapshotTX *storage.MockReadOnlyAdminTX
+	server     *adminServer
+}
+
+// setupAdminStorage configures storage mocks according to input parameters.
```

Comment:
Comment doesn't match function name; I'd lean towards changing the function to match the comment so this fn. has a different name than the one in `integration/admin/`.